src/System.Text.Json/src/System/Text/Json/JsonConstants.cs

```diff
@@ -72,6 +72,11 @@ internal static class JsonConstants
         public const int MaxDateTimeUtcOffsetHours = 14; // The UTC offset portion of a TimeSpan or DateTime can be no more than 14 hours and no less than -14 hours.
         public const int DateTimeNumFractionDigits = 7;  // TimeSpan and DateTime formats allow exactly up to many digits for specifying the fraction after the seconds.
         public const int MaxDateTimeFraction = 9_999_999;  // The largest fraction expressible by TimeSpan and DateTime formats
+        public const int DateTimeParseNumFractionDigits = 15; // The maximum number of fraction digits the Json DateTime parser allows
+        public const int MaximumDateTimeOffsetParseLength = (MaximumFormatDateTimeOffsetLength +
+            (DateTimeParseNumFractionDigits - DateTimeNumFractionDigits)); // Like StandardFormat 'O' for DateTimeOffset, but allowing 8 additional (up to 15) fraction digits.
+        public const int MinimumDateTimeParseLength = 10; // YYYY-MM-DD
+        public const int MaximumEscapedDateTimeOffsetParseLength = MaxExpansionFactorWhileEscaping * MaximumDateTimeOffsetParseLength;
```

Comment:
33 + 8 = 41
41 * 6 = 246

So, we could move `DateTimeParseNumFractionDigits` to 16 and still be < the next power of 2 (i.e .256)